### DISCUSSION TITLE
Compile Wasm project targetting NET7 using dotnet build

### DISCUSSION THREAD

**Discussion by giraudremy** [unoplatform/uno#9205] [Q&A] [answered]
Hi,

I'm trying to compile Wasm projet targetting NET7 by using :

> dotnet build

and I have this error :

> C:\Users\Patachou\.nuget\packages\uno.sourcegenerationtasks\4.0.2\build\netstandard1.0\Uno.SourceGenerationTasks.targets(134,2): error : Unable to find Uno.SourceGeneration.Host.dll (in C:\Users\Patachou\.nuget\packages\uno.sourcegenerationtasks\4.0.2\build\Dev15.0\netcoreapp3.1\..\..\..\..\Uno.SourceGeneration.Host\bin\Release\net7 or C:\Users\Patachou\.nuget\packages\uno.sourcegenerationtasks\4.0.2\build\Dev15.0\netcoreapp3.1\..\..\host\net7) [C:\Users\Patachou\source\repos\UnoNet7\UnoNet7\UnoNet7.Wasm\UnoNet7.Wasm.csproj]
>     0 Warning(s)
>     1 Error(s)

Do I need to pass other information to dotnet?

Changing 
`<TargetFramework>net7.0</TargetFramework> `
to 
`<TargetFramework>net6.0</TargetFramework>`
and dotnet build work well
[UnoNet7.zip](https://github.com/unoplatform/uno/files/9039752/UnoNet7.zip)

**Comment by ebariche** [comment]
Hi !

We don't support the net7.0 TFM for WASM projects yet ! You can use net6.0 in the meantime.

Unintuitively, the latest bootstrapper will still use NET 7 so you can play around with it : 

![image](https://user-images.githubusercontent.com/33458222/177193315-fb8c32d3-5786-412b-813b-e18ae7f155a3.png)

I'm going to open an issue to have Uno.SourceGeneration updated to NET 7 !

**Reply by ebariche** [reply]
Issue : #9207

**Comment by giraudremy** [comment]
Thank for reply. I have issue with AOT and net6.0 framework, and it solved with net7.0, that why I want to use NET7

**Reply by francoistanguay** [reply]
What is your AOT issue that got fixed with net7? Existing Uno issued tracked in our repo? Something specific to your app?

**Reply by giraudremy** [reply]
Here we can see issue on my application when I activate AOT with NET6
[https://github.com/unoplatform/uno/issues/8741#issuecomment-1125388339](url)

With NET7 my application now work

**Comment by spouliot** [comment, marked as answer]
@giraudremy trying the workaround from https://github.com/unoplatform/uno/issues/9207#issuecomment-1175214475 should, hopefully, help you get past this build error

**Reply by giraudremy** [reply]
Thank for the workaround, I have duplicate the directory on Windows and It's work
